tests/chainer_tests/test_link.py

```diff
@@ -471,10 +471,19 @@ def test_update_enabled(self):
         self.assertTrue(self.link.update_enabled)
 
     def test_count_params(self):
-        self.assertEqual(self.link.count_params(), 8)
         with warnings.catch_warnings(record=True) as w:
             warnings.simplefilter('always')
             self.link.count_params()
+        assert len(w) == 2
+        assert w[0].category is UserWarning
+        assert self.link.count_params() == 8
+
+        self.link.u.initialize((2, 3))
+        self.link.v.initialize((2, 3))
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter('always')
+            self.link.count_params()
+            assert len(w) == 0
```

Comment:
You can check the condition simply with `assert not w` because w is a (empty) list 
(cf. https://docs.python.org/3.6/library/warnings.html#warnings.catch_warnings)

Reply:
But current code is also OK as it is easier to understand.